homeassistant/components/light/__init__.py

```diff
@@ -148,6 +148,16 @@ def preprocess_turn_on_alternatives(params):
         params.setdefault(ATTR_XY_COLOR, profile[:2])
         params.setdefault(ATTR_BRIGHTNESS, profile[2])
 
+    brightness_pct = params.pop(ATTR_BRIGHTNESS_PCT, None)
+    if brightness_pct is not None:
+        params[ATTR_BRIGHTNESS] = int(255 * brightness_pct/100)
+
+    if ATTR_BRIGHTNESS in params and params[ATTR_BRIGHTNESS] == 0:
+        # Zero brightness: Light will be turned off
+        for k in list(params.keys()):
+            params = {k: v for k, v in params.items() if k in [ATTR_TRANSITION,
```

Comment:
You will need to add a test so that you will notice that this won't work, as `preprocess_turn_on_alternatives` currently does not have a return value

Reply:
Wow, that was stupid..
Fixed, and tests updated.